logging/src/logging.cc

```diff
@@ -34,38 +33,61 @@ namespace leatherman { namespace logging {
     static bool g_colorize = false;
     static bool g_error_logged = false;
 
+    class color_writer : public sinks::basic_sink_backend<sinks::synchronized_feeding>
+    {
+     public:
+        color_writer(ostream *dst);
+        void consume(boost::log::record_view const& rec);
+     private:
+        ostream *_dst;
```

Comment:
It looks like the synchronous_sink class has some sort of requirement on the backend it wraps that ostream& violates.

```
In file included from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/log/sinks/sync_frontend.hpp:30:0,
                 from C:\Users\lasth\Downloads\facter\vendor\leatherman\logging\src\logging.cc:14:
C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/smart_ptr/make_shared_object.hpp: In instantiation of 'typename boost::detail::sp_if_not_array<T>::type boost::make_shared(Args&& ...) [with T = leatherman::logging::color_writer; Args = {const std::basic_ostream<char, std::char_traits<char> >&}; typename boost::detail::sp_if_not_array<T>::type = boost::shared_ptr<leatherman::logging::color_writer>]':
C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/log/sinks/sync_frontend.hpp:119:5:   required from 'boost::log::v2s_mt_nt5::sinks::synchronous_sink<SinkBackendT>::synchronous_sink(const T0&) [with T0 = std::basic_ostream<char>; SinkBackendT = leatherman::logging::color_writer]'
C:\Users\lasth\Downloads\facter\vendor\leatherman\logging\src\logging.cc:74:54:   required from here
C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/smart_ptr/make_shared_object.hpp:254:5: error: no matching function for call to 'leatherman::logging::color_writer::color_writer(const std::basic_ostream<char>&)'
     ::new( pv ) T( boost::detail::sp_forward<Args>( args )... );
     ^
C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/smart_ptr/make_shared_object.hpp:254:5: note: candidate is:
C:\Users\lasth\Downloads\facter\vendor\leatherman\logging\src\logging.cc:44:5: note: leatherman::logging::color_writer::color_writer(std::ostream&)
     color_writer::color_writer(ostream &dst) : _dst(dst) {}
     ^
C:\Users\lasth\Downloads\facter\vendor\leatherman\logging\src\logging.cc:44:5: note:   no known conversion for argument 1 from 'const std::basic_ostream<char>' to 'std::ostream& {aka std::basic_ostream<char>&}'
In file included from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/system/system_error.hpp:14:0,
                 from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/thread/exceptions.hpp:22,
                 from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/thread/win32/thread_primitives.hpp:17,
                 from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/thread/win32/basic_recursive_mutex.hpp:13,
                 from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/thread/win32/recursive_mutex.hpp:13,
                 from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/thread/recursive_mutex.hpp:14,
                 from C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/log/sinks/sync_frontend.hpp:31,
                 from C:\Users\lasth\Downloads\facter\vendor\leatherman\logging\src\logging.cc:14:
C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/system/error_code.hpp:221:36: error: 'boost::system::posix_category' defined but not used [-Werror=unused-variable]
     static const error_category &  posix_category = generic_category();
                                    ^
C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/system/error_code.hpp:222:36: error: 'boost::system::errno_ecat' defined but not used [-Werror=unused-variable]
     static const error_category &  errno_ecat     = generic_category();
                                    ^
C:/Users/lasth/Downloads/deps/include/boost-1_57/boost/system/error_code.hpp:223:36: error: 'boost::system::native_ecat' defined but not used [-Werror=unused-variable]
     static const error_category &  native_ecat    = system_category();
                                    ^
cc1plus.exe: all warnings being treated as errors
make[2]: *** [logging/CMakeFiles/leatherman_logging.dir/src/logging.cc.obj] Error 1
```